src/Module/Markdown/Step/ProcessMarkdownFileName.php

```diff
@@ -36,7 +42,13 @@ private function renameFileExtension(MarkdownFile $file)
 
     private function renameReadme(MarkdownFile $file)
     {
-        if ($file->getBasename() !== 'README.html') {
+
+        if(
+          !empty($this->project->metadata['index']) &&
+          $file->getBasename() !== $this->replaceExtension(basename($this->project->metadata['index'])))
+        {
+          return;
+        } else if(empty($this->project->metadata['index']) && $file->getBasename() !== 'README.html') {
```

Comment:
It might be clearer to use a variable. For example:

```php
$indexFile = empty($this->project->metadata['index']) ? 'README.md' : $this->project->metadata['index'];
$indexFile = $this->replaceExtension(basename($indexFile));
if ($file->getBasename() !== $indexFile) {
```